src/RefractDataStructure.cc

```diff
@@ -900,13 +900,13 @@ namespace drafter {
                 else if (ValueHasName(input.node) || ValueHasMembers(input.node)) {
                     return Trait::template Invoke<refract::ObjectElement>(input, defaultNestedType);
                 }
-
-                return MsonMemberToRefract<Trait>(input, defaultNestedType, defaultNestedType);
+                else if (nameType != defaultNestedType) {
+                    return MsonMemberToRefract<Trait>(input, defaultNestedType, defaultNestedType);
+                }
             }
-
-            default:
-                throw snowcrash::Error("unknown type of mson member", snowcrash::MSONError);
         }
+
+        throw snowcrash::Error("unknown type of mson member", snowcrash::MSONError);
```

Comment:
I didn't find any other tests for `unknown type of moon member`, unsure how we can go about testing this. Would love suggestions.